### DISCUSSION TITLE
Jupyter Notebook rendering

### DISCUSSION THREAD

**Discussion by pseusys** [community/community#3279] [Mobile]
Mobile app can not render Jupyter Notebook files, it demonstrates them as raw json-text file in browse code mode.
Maybe it could be possible to add this feature to mobile app as it was already done in github web site?
It could've come in handy.

**Comment by BunnyNoBugs** [comment]
Yeah, would totally agree! Many data science related repositories mainly consist of `.ipynb` files and current Jupyter Notebook support in the mobile app makes browsing these essentially meaningless. Would really appretiate proper Notebook viewing instead of raw `.json` files

**Reply by ogtega** [reply]
It's an iframe on the GitHub website, so a webview would do just fine.

**Comment by ghost** [comment]
On safari/chrome ios apps I do see the rendered stuff which is great, but I have a smallish phone and the cells get cut off and I cant scroll or zoom out which is a little unfortunate. Maybe I'm missing something obvious.

![IMG_0811](https://user-images.githubusercontent.com/48896686/232964027-d05d82d3-b5da-4099-83dd-4455b20c484c.png)

**Comment by eddiegordo0** [comment]
`
![logo_android_auto_color_2x_web_512dp original](https://github.com/community/community/assets/67650427/99bc636e-edd7-4058-bcc1-1e608d54e33a)

`

**Comment by Tolgahan0** [comment]
Enhancing Jupyter Notebook Rendering on GitHub: A Better Experience for Data Scientists

Data scientists, analysts, and researchers rely heavily on Jupyter Notebooks for their data exploration, analysis, and visualization needs. These notebooks have become the backbone of collaborative and reproducible research. GitHub has long been a popular platform for sharing code, collaborating on projects, and discussing ideas. To provide an even more seamless experience for the data science community, GitHub has been actively working to enhance Jupyter Notebook rendering. In this post, we'll delve into the improvements made and their impact on the way you share and interact with Jupyter Notebooks on GitHub.

Richer Rendering:
GitHub has embraced the importance of rich rendering for Jupyter Notebooks. With recent updates, notebooks shared on the platform now offer an improved rendering of visualizations, charts, and interactive widgets. Whether your notebook includes intricate visualizations, data-driven plots, or interactive maps, viewers can now explore your work with greater clarity and interactivity. This enhancement not only elevates the overall aesthetics of your content but also fosters a deeper understanding of your data-driven insights.

Seamless Interaction:
Interactivity is a crucial aspect of Jupyter Notebooks. GitHub's focus on seamless interaction means that viewers can now experience the interactive elements in their notebooks more intuitively. This includes sliders, dropdowns, and other widgets that allow users to tweak parameters and witness real-time changes in visualizations. This improved interactivity empowers both authors and audiences to experiment, validate hypotheses, and grasp complex concepts more effectively.

Versioned Outputs:
Version control lies at the heart of GitHub's functionality. Recognizing this, Jupyter Notebook rendering enhancements also extend to versioned outputs. Now, when changes are made to your notebook, viewers can not only access the latest version but also explore historical outputs. This is particularly valuable for tracking the evolution of data visualizations and ensuring the reproducibility of results over time.

Collaborative Iteration:
Collaboration is a cornerstone of GitHub, and the recent enhancements to Jupyter Notebook rendering amplify this aspect. Collaborators can now offer feedback, suggest improvements, and engage in meaningful discussions directly within the notebook environment. This fosters a more integrated and iterative collaborative process, enabling data scientists to refine their analyses based on real-time feedback from peers.

Conclusion:
The strides taken by GitHub in enhancing Jupyter Notebook rendering underscore the platform's commitment to providing a robust and engaging experience for data scientists and researchers. From crisper visualizations to improved interactivity and versioned outputs, these updates elevate the way we share, collaborate, and iterate on data-driven projects. As the data science community continues to grow, these enhancements play a pivotal role in facilitating insightful discussions and pushing the boundaries of collaborative research. So, whether you're sharing your latest data analysis or delving into someone else's work, the improved Jupyter Notebook rendering on GitHub promises a more immersive and rewarding experience for all.
